Support packet multiplicity in frames_processed (#307)
quicwg/qlog: draft-ietf-quic-qlog-quic-events.md

In some implementations, it can be difficult to log frames directly, even
when using packet_sent and packet_received events. For these cases, this event

> also contains the packet_number field, which can be used to more explicitly

Reviewer: ```suggestion
also contains the packet_numbers field, which can be used to more explicitly
```

link this event to the packet_sent/received events. The field is an array, which
supports using a single "frames_processed" event for multiple frames received
over multiple packets. To map between frames and packets, the position and order

> of entries in the "frames" and "packet_number" is used. If the "packet_number"

Reviewer: ```suggestion
of entries in the "frames" and "packet_numbers" is used. If the optional "packet_numbers"
```

field is used, each frame MUST have a corresponding packet number at the same
index.

Definition:

~~~ cddl
QUICFramesProcessed = {
    frames: [* $QuicFrame]

>     ? packet_number: [* uint64]

Reviewer: ```suggestion
    ? packet_numbers: [* uint64]
```

For consistency with existing events like `packets_acked` and partially the `datagram_*` events that talk about `datagram_ids` not `datagram_id`

}
~~~
{: #quic-framesprocessed-def title="QUICFramesProcessed definition"}

For example, an instance of this event that represents four STREAM frames
received over two packets would have the fields serialized as:

~~~
"frames":[

>   {"frame_type":"stream","stream_id":0,"offset":0,"length":1},
>   {"frame_type":"stream","stream_id":0,"offset":1,"length":1},
>   {"frame_type":"stream","stream_id":1,"offset":0,"length":1},
>   {"frame_type":"stream","stream_id":1,"offset":1,"length":1}

Reviewer: ```suggestion
  {"frame_type":"stream","stream_id":0,"offset":0,"length":500},
  {"frame_type":"stream","stream_id":0,"offset":500,"length":200},
  {"frame_type":"stream","stream_id":1,"offset":0,"length":300},
  {"frame_type":"stream","stream_id":1,"offset":300,"length":50}
```
Making the number more different between the two streams to more clearly differentiate them + make the example slightly more realistic.

  ],

> "packet_number":[

Reviewer: ```suggestion
"packet_numbers":[
```
